Tasks/Common/azure-arm-rest/azure-arm-app-service.ts

```diff
@@ -278,15 +311,39 @@ export class AzureAppService {
         }
     }
 
-    public async patchConfiguration(properties): Promise<void> {
-        var applicationSettings = await this.getConfiguration();
+    public async patchMetaData(properties): Promise<void> {
+        var applicationSettings = await this.getMetaData();
         for(var key in properties) {
             applicationSettings.properties[key] = properties[key];
         }
 
-        await this.updateConfiguration(applicationSettings);
+        await this.updateMetaData(applicationSettings);
+    }
+
+    public async updateMetaData(applicationSettings): Promise<AzureAppServiceConfigurationDetails> {
+        try {
+            var httpRequest = new webClient.WebRequest();
+            httpRequest.method = 'PUT';
+            httpRequest.body = JSON.stringify(applicationSettings);
+            var slotUrl: string = !!this._slot ? `/slots/${this._slot}` : '';
+            httpRequest.uri = this._client.getRequestUri(`//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/${slotUrl}/config/metadata`,
```

Comment:
P2: wont it result in two // in the URL if the slot is empty.

Reply:
yes, it results in two //, but it didnt throw any error from API side.